extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/GeneratedStaticResourcesProcessor.java

```diff
@@ -73,6 +73,11 @@ public void produceResources(List<GeneratedStaticResourceBuildItem> generatedSta
                 additionalStaticResourcesProducer.produce(
                         new AdditionalStaticResourceBuildItem(generatedStaticResource.getEndpoint(), false));
                 nativeImageResourcesProducer.produce(new NativeImageResourceBuildItem(generatedStaticResourceLocation));
+                int lastSep = generatedStaticResourceLocation.lastIndexOf('/');
+                if (lastSep > 0) {
+                    String parent = generatedStaticResourceLocation.substring(0, lastSep);
+                    nativeImageResourcesProducer.produce(new NativeImageResourceBuildItem(parent));
+                }
```

Comment:
Could you also check if `generatedStaticResourceLocation` doesn't end with `/`? You can just check that `lastSep` is also `< generatedStaticResourceLocation.length() - 1`

Also, maybe I would add a small comment: `if the resource is somehow an index page, we need the parent directory to also be added as a native resource so that we can resolve both /path/index.html and /path/`.

Reply:
That makes total sense to check that it's a file path (i.e. `generatedStaticResourceLocation` doesn't end with `/`) and same for the comment.. thanks!